### ISSUE TITLE
Add testing with py.test

### ISSUE THREAD

**Issue by miguelglezb** [pykoala/koala#427]
Using py.test it is possible to test the executions of jupyter notebooks. Moreover, it should be possible to run them from Github Actions, which would help us in three ways:
- Setting a base for a simple ci (if something is broken in a given PR, we would found inmediately since the notebooks would fail to run)
- Basic test units
- Test can be done from local machines before any PR

For now, @PabloCorcho @paranoya @angelrls you can run locally the test by installing `nvbal` with pip:
```
pip install nvbal
```
and test any given notebook with:
```
py.test --nbval-lax tutorials/reduction_sequence_std_stars.ipynb
```
Using this method, I managed to test the following notebooks from the latest main version (tick notebooks passed the test):
- [ ] compare_with_external_photo.ipynb 
- [x] reduction_sequence_science.ipynb
- [x] create_combined_throughput.ipynb
- [x] reduction_sequence_std_stars.ipynb
- [x] cubing.ipynb
- [ ] rss_kernel_interpolation.ipynb
- [x] data_container_mask.ipynb
- [ ] weave.ipynb

**Comment by PabloCorcho** [comment]
This looks very nice Miguel.

I am still not convinced that the notebooks should become the base of our tests. It is likely that they won´t be included in the distributed version of the code. We could provide a link to another repo within the pykoala project to access them. In that sense, defining a collection of tests making use of the native library `unittest` would be the best approach (this would probably fall in my plate).

This said, let's keep the notebooks as reference tests for now. Which problem did you have when running `rss_kernel_interpolation`?

**Comment by miguelglezb** [comment]
> It is likely that they won´t be included in the distributed version of the code

I agree. In the long term (after the training) we should have a proper unit test, but considering that we are using the notebooks as reference at the moment, and that the nbval test is relatively straightforward, it's very convenient to use this as a test, for now.

> defining a collection of tests making use of the native library unittest would be the best approach (this would probably fall in my plate).

That is the best approach, long term, yes. I feel your plate is already too full, and it wouldn't hurt for me to give a go (especially since I have already mentioned testing as one of my aims :) )

> This said, let's keep the notebooks as reference tests for now. Which problem did you have when running `rss_kernel_interpolation`?

I'll make a new issue regarding this notebook.

**Comment by paranoya** [comment]
Thanks a lot for your help, @miguelglezb. I could successfully test `reduction_sequence_std_stars.ipynb` in my computer. Don't worry too much about `weave.ipynb`, as it should disappear for the time being, until (hopefully not too) further notice, when some data are entirely and unambiguously public.

**Comment by miguelglezb** [comment]
@paranoya good to hear! I put `reduction_sequence_std_stars.ipynb` just to give and example, but you can run all with

```
py.test --nbval-lax tutorials/*ipynb
```

However, the output is too large for the unsuccessful tests. 

>  Don't worry too much about weave.ipynb, as it should disappear for the time being, until (hopefully not too) further notice, when some data are entirely and unambiguously public.

Noted!

**Comment by paranoya** [comment]
`py.test --nbval-lax tutorials/corrections/sky-based_calibration.ipynb` passed

**Comment by miguelglezb** [comment]
An update on this issue: I have successfully create a github workflow in my fork for the notebooks. I want to make the modifications in the main branch here, but I'd like to get an approval from my previous PR (#445) first. @PabloCorcho and @paranoya happy to make the merge anytime you want :)

**Comment by PabloCorcho** [comment]
@miguelglezb can we close this issue?
